libs/service/api_service.cpp

```diff
@@ -80,11 +64,40 @@ _build_helper_id_to_address_map(
         return win32_error_code_to_ebpf_result(result);
     }
 
-    helper_addresses.resize(helper_id_mapping.size());
-
     index = 0;
-    for (auto& address : helper_addresses) {
-        address = reply->address[index++];
+    for (auto& [helper_id, helper_address] : helper_id_to_address) {
+        helper_address = reply->address[index++];
+    }
+
+    return EBPF_SUCCESS;
+}
+
+static ebpf_result_t
+_build_helper_id_to_address_map(
+    _In_reads_(instruction_count) ebpf_inst* instructions,
+    uint32_t instruction_count,
+    const std::map<uint32_t, uint64_t>& helper_id_to_address,
+    uint32_t& unwind_index)
+{
+    // Note:
+    // eBPF supports helper IDs in the range [1, MAXUINT32]
+    // uBPF jitter only supports helper IDs in the range [0,63]
+    // Build a table to map [1, MAXUINT32] -> [0,63]
+    std::map<uint32_t, uint32_t> helper_id_mapping;
+    unwind_index = MAXUINT32;
+
+    uint32_t index = 0;
+    for (auto [helper_id, helper_address] : helper_id_to_address) {
+        helper_id_mapping[helper_id] = index++;
+    }
+
+    if (helper_id_mapping.size() == 0) {
+        return EBPF_SUCCESS;
+    }
+
+    // uBPF jitter supports a maximum of 64 helper functions
+    if (helper_id_mapping.size() > 64) {
```

Comment:
does uBPF really provide no define macro that can be used instead of hard coding 64 here? 
@Alan-Jowett  If it doesn't then we should file an issue against uBPF.
Looks like right now it's an internal-only define:
```
vm\ubpf_vm.c:#define MAX_EXT_FUNCS 64
```
Either way, we should use a macro in our repo and not hard code 64 here.
So for now do something in this file like:
```
// TODO(<insert ubpf issue number here>): uBPF has a max number of helper functions hard coded,
// but doesn't expose the define to callers, so we define it here.
#define UBPF_MAX_EXT_FUNCS 64
```

Reply:
Added.